apps/arweave/src/ar_http_iface_middleware.erl

```diff
@@ -273,7 +273,8 @@ handle(<<"POST">>, [<<"wallet">>], Req) ->
 handle(<<"POST">>, [<<"tx">>], Req) ->
 	TXJSON = ar_http_req:body(Req),
 	TX = ar_serialize:json_struct_to_tx(TXJSON),
-	case handle_post_tx(TX) of
+	{PeerIP, _Port} = cowboy_req:peer(Req),
```

Comment:
Is there a way to move this up to the top of the function, so we don't unnecessarily read the body and attempt to parse it as JSON before performing the next actions?

Reply:
The body is currently parsed in a separate middleware. There is a WIP [branch](https://github.com/ArweaveTeam/arweave/pull/187) changing it, one of the prerequisites for "fast write".